ROCANA-9215: Rename and change package of Lucene forked code

bradcupit opened the pull request.

Commit:
ROCANA-9215: Rename and change package of Lucene forked code

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaBlockTreeTermsReader.java:
```diff
@@ -44,7 +45,28 @@
+ *
+ * What changed in the fork?
+ *   - Commented out the constructor's call to {@link CodecUtil#checksumEntireFile(IndexInput)}
+ *     for performance.
```
Maybe add: "for performance improvements and on HDFS this is a redundant check to the checksumming of blocks already done by HDFS".

bradcupit replied:
changed the wording. I expanded on it quite a bit. I encourage you to verify my wording is correct

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaBlockTreeTermsReader.java:
```diff
@@ -44,7 +45,28 @@
+ *   - Changed these javadocs.
+ *
+ * To see a full diff of changes: compare this version to this file's initial commit (which
+ * is the original file from Lucene).
```
should add the version and/or Lucene tag to this comment: 5.5.0, git tag `releases/lucene-solr/5.5.0`

bradcupit replied:
done, but at the top of the javadocs instead

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaBlockTreeTermsReader.java:
```diff
@@ -44,7 +45,28 @@
+ * To see a full diff of changes: compare this version to this file's initial commit (which
+ * is the original file from Lucene).
+ *
+ * See ROCANA-8229.
```
This JIRA is not publicly visible but this codec is open sourced, so shouldn't refer to it here, but rather describe the information here.

bradcupit replied:
I'll remove it from all the forked files. For the benefit of the Rocana engineer (and let's be honest, there's a 90% chance this code's only ever be used by Rocana engineers) I'll keep the reference in the main RocanaSearchCodecV1 javadocs, and say what we have in the README.md:

> If you have access to Rocana's JIRA, see ROCANA-8229.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaBlockTreeTermsReader.java:
```diff
@@ -44,7 +45,28 @@
 
-/** A block-based terms index and dictionary that assigns
+/**
+ * Fork of Lucene's {@link com.rocana.lucene.codec.v1.RocanaBlockTreeTermsReader}.
```
This reference is wrong - should point to Lucene's package/class.

bradcupit replied:
nice catch! Eclipse did this and apparently I fixed some but not all

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaBlockTreeTermsReader.java:
```diff
@@ -228,8 +250,8 @@ public BlockTreeTermsReader(PostingsReaderBase postingsReader, SegmentReadState
-        FieldReader previous = fields.put(fieldInfo.name,       
-                                          new FieldReader(this, fieldInfo, numTerms, rootCode, sumTotalTermFreq, sumDocFreq, docCount,
+        RocanaFieldReader previous = fields.put(fieldInfo.name,       
+                                          new RocanaFieldReader(this, fieldInfo, numTerms, rootCode, sumTotalTermFreq, sumDocFreq, docCount,
```
The spacing formatting doesn't conform to Rocana standards. Do we apply those standards to open sourced code? (I vote yes.)

bradcupit replied:
We're an extremely minor fork of Lucene, ultimately just trying to comment out one line of code.

Ideally we'd stay as close to the Lucene code as possible so we can diff to see **exactly** what changed in our fork. I've used this style before and it's very nice. Even if we never use it we have a scientific way of determining exactly what we changed, no matter how much or little we diverge in the future. We could also generate a patch from this diff, to apply onto a newer version of Lucene. I'm not recommending we do that, just saying it's possible.

I vote we don't adhere to Rocana's standard as that would muddy the diff.

mpeterson0x0 replied:
OK, we can change it later if we need to do additional changes.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaFieldReader.java:
```diff
@@ -36,15 +36,36 @@
 import org.apache.lucene.util.fst.FST;
 
 /**
+ * Fork of Lucene's {@link com.rocana.lucene.codec.v1.RocanaFieldReader}.
```
Wrong reference - should point to Lucene's package/class.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaFieldReader.java:
```diff
@@ -36,15 +36,36 @@
+ * class and override just the methods we need to change. Unfortunately
+ * in most cases that wasn't an option because many Lucene classes are final.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaIntersectTermsEnum.java:
```diff
@@ -35,7 +35,29 @@
+ * in most cases that wasn't an option because many Lucene classes are final.
+ *
+ * See ROCANA-8229
+ * @see RocanaSearchCodecV1
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaIntersectTermsEnumFrame.java:
```diff
@@ -27,8 +27,29 @@
+ * class and override just the methods we need to change. Unfortunately
+ * in most cases that wasn't an option because many Lucene classes are final.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaLucene50PostingsFormat.java:
```diff
@@ -40,6 +41,25 @@
+ * To see a full diff of changes: compare this version to this file's initial commit (which
+ * is the original file from Lucene).
+ *
+ * See ROCANA-8229.
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaLucene50PostingsFormat.java:
```diff
@@ -40,6 +41,25 @@
 import org.apache.lucene.util.packed.PackedInts;
 
 /**
+ * Fork of Lucene's {@link com.rocana.lucene.codec.v1.RocanaLucene50PostingsFormat}.
```
wrong reference

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaSegmentTermsEnum.java:
```diff
@@ -33,19 +33,41 @@
+ * class and override just the methods we need to change. Unfortunately
+ * in most cases that wasn't an option because many Lucene classes are final.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaSegmentTermsEnumFrame.java:
```diff
@@ -27,7 +27,28 @@
+ * class and override just the methods we need to change. Unfortunately
+ * in most cases that wasn't an option because many Lucene classes are final.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaStats.java:
```diff
@@ -28,11 +28,36 @@
 import org.apache.lucene.util.IOUtils;
 
 /**
+ * Fork of Lucene's {@link com.rocana.lucene.codec.v1.RocanaStats}.
```
wrong reference

mpeterson0x0 commented on src/main/java/com/rocana/lucene/codec/v1/RocanaStats.java:
```diff
@@ -28,11 +28,36 @@
+ * performance path, we don't want to pay the performance price of using
+ * reflection to access that private field.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat.java:
```diff
@@ -31,10 +31,24 @@
 import org.apache.lucene.util.TestUtil;
 
 /**
+ * Fork of Lucene's {@link com.rocana.lucene.codec.v1.TestBlockPostingsFormat}.
```
wrong reference

mpeterson0x0 commented on src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat.java:
```diff
@@ -31,10 +31,24 @@
+ *   - Removed trailing whitespace.
+ *   - Changed these javadocs.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat2.java:
```diff
@@ -32,8 +32,22 @@
-/** 
- * Tests special cases of BlockPostingsFormat 
+/**
+ * Fork of Lucene's {@link com.rocana.lucene.codec.v1.TestBlockPostingsFormat2}.
```
wrong reference

mpeterson0x0 commented on src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat2.java:
```diff
@@ -32,8 +32,22 @@
+ *   - Removed trailing whitespace.
+ *   - Changed these javadocs.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented on src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat3.java:
```diff
@@ -56,12 +56,26 @@
-/** 
- * Tests partial enumeration (only pulling a subset of the indexed data) 
+/**
+ * Fork of Lucene's {@link com.rocana.lucene.codec.v1.TestBlockPostingsFormat3}.
```
wrong reference

mpeterson0x0 commented on src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat3.java:
```diff
@@ -56,12 +56,26 @@
+ *   - Removed trailing whitespace.
+ *   - Changed these javadocs.
+ *
+ * See ROCANA-8229
```
Same comment here about referring to internal only JIRA tickets in open sourced code.

mpeterson0x0 commented:
Overall comment: how do we know that you caught all the references and properly renamed them? One way: check all the import statements (either manually or write a script) to make sure none of the forked classes are referenced in the forked classes or anywhere else in the Rocana codebase - should we add that as an acceptance criterion for this ticket?

mpeterson0x0 reviewed (changes requested):
Finished round 1 review. Minor javadoc changes requested and then one overall sanity check (see overall comment) requested.

bradcupit commented:
Regarding the overall comment:
>  how do we know that you caught all the references and properly renamed them?

Fair question.

Eclipse renamed them and renamed them in the files. That's why so many of the javadoc references were wrong (the ones that were correct I fixed manually). Also, the below forked classes are package private. Since we're in a different package we'd get a compile error if we accidentally referenced these:
* IntersectTermsEnum
* IntersectTermsEnumFrame
* SegmentTermsEnum
* SegmentTermsEnumFrame

I verified we don't use any of the files using a regular expression. For example:
```[^a]BlockTreeTermsReader```
That finds any reference that's not Rocan**aBlockTreeTermsReader**. In this case it found two hits: one in the javadocs where I reference the original Lucene file name, and a comment in another file.

I fixed the comments just in case we ever uncomment that code.

I repeated this process for each file and there are no violations other than these two intentional ones:
* In TestBlockPostingsFormat.testFinalBlock() line 72 returns a FieldReader, not a RocanaFieldReader. We get a ClassCastException if we try to use RocanaFieldReader.
* In the same test method since we're using FieldReader.getStats() we'll get a Stats, not RocanaStats.

Commit:
PR feedback + some documentation changes

mpeterson0x0 reviewed (approved):
+1
